numpy/core/tests/test_numeric.py

```diff
@@ -48,6 +48,14 @@ def test_reshape_from_zero(self):
         assert_array_equal(Ar, np.zeros((2, 1), Ar.dtype))
         assert_equal(A.dtype, Ar.dtype)
 
+    def test_negativeresize(self):
+        A = np.arange(0, 10, dtype=np.float32)
+        new_shape = (10,-1)
```

Comment:
Can you add a test for `(-10, -1)` too, which should also fail?

(hint: It won't fail, you need to adjust your condition above)

Reply:
I think this should be resolved now